index.bs

```diff
@@ -736,6 +736,16 @@ Each {{XRSpace}} has a <dfn for="XRSpace">native origin</dfn> that is tracked by
 
 The [=effective origin=] of an {{XRSpace}} can only be observed in the coordinate system of another {{XRSpace}} as an {{XRPose}}, returned by an {{XRFrame}}'s {{XRFrame/getPose()}} method. The spatial relationship between {{XRSpace}}s MAY change between {{XRFrame}}s.
 
+Each {{XRSpace}} has an <dfn for="XRSpace">internal type</dfn>, used to internally distinguish the various kinds of spaces. This value can be <dfn for="XRSpace">viewer space</dfn>, <dfn for="XRSpace">input target ray</dfn>, <dfn for="XRSpace">input grip</dfn>, or one of the variants of {{XRReferenceSpaceType}}.
+
+An {{XRSpace}} of type [=XRSpace/input target ray=], [=XRSpace/input grip=] must have an associated {{XRInputSource}} <dfn for="XRSpace">input source</dfn>.
+
+The [=native origin=] of an {{XRSpace}} with [=XRSpace/internal type=] of [=XRSpace/viewer space=] is the position and orientation of the [=viewer=].
+
+The [=native origin=] of an {{XRSpace}} with [=XRSpace/internal type=] of [=XRSpace/input target ray=] is the position and orientation of the preferred pointing ray of its [=XRSpace/input source=], as defined by the {{targetRayMode}}.
+
+The [=native origin=] of an {{XRSpace}} with [=XRSpace/internal type=] of [=XRSpace/input grip=] is the position and orientation that should be used to render virtual objects corresponding to its [=XRSpace/input source=] such that they appear to be held in the user's hand. If the user were to hold an [=XRSpace/input source=] that is straight rod, this [=native origin=] has a position at the centroid of their curled fingers. The <code>-Z</code> axis of the orientation points along the length of the rod towards their thumb. The <code>X</code> axis is perpendicular to the back of the hand being described, with back of the users right hand pointing towards <code>+X</code> and the back of the user's left hand pointing towards <code>-X</code>. The <code>Y</code> axis is implied by the relationship between the <code>X</code> and <code>Z</code> axis, with <code>+Y</code> roughly pointing in the direction of the user's arm.
```

Comment:
If we were to define this relative to a common "native origin" for an XRInputSource, most of this paragraph would be describing the originOffset